app/javascript/react/screens/App/common/commonUtilitiesHelper.js

```diff
@@ -0,0 +1,18 @@
+const commonUtilitiesHelper = {
+  getMostRecentEntityByCreationDate: entities =>
+    entities.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current)),
+
+  groupBy: (items, key) =>
+    items.reduce(
+      (result, item) => ({
+        ...result,
+        [item[key]]: [...(result[item[key]] || []), item]
+      }),
+      {}
+    ),
+
+  flattenArray: (arrayItems, flattenedArray) =>
```

Comment:
wondering why not use https://lodash.com/docs/4.17.4#flatten?  We have already use lodash in the app 🤔

Reply:
I'm going to have to respectfully disagree here. I think it's preferred to use es6 methods when the functions are trivial and there is not much more to implement using a native function. Individual Lodash imports will soon be stripped by Webpack tree shaking techniques and loaded individually, however I still don't think that's a reason to prefer them (since they often times add more code on the loading end). In these cases I prefer to extend the native functions which removes the need for those imports. I have seen cases where lodash functions do provide a lot of value (`sortBy` for example is non trivial), but these are specific to the use case...

Reply:
I will believe this was mentioned in several Foreman PRs as well...

Reply:
Ooooo! We disagree indeed. I don't have performance metrics to compare, so I cannot assert one method is "better than another"  🤷‍♂️  *BUT* the less code we have to maintain, watch something go wrong with, the better. Both of these functions have been written before, I see no strong case for us to rewrite them.

☝️ being said, this is a blocker, my personal preference to not rewrite commonly functions will not hinder approval of this

Reply:
Yea, obviously we are still loading Lodash and not stripping it...so it's already included and would be minimal impact on this project, but long term I still think it's considered bad practice now by most communities (kind of like using jQuery at this point, we avoid it when we can but obviously we still rely on it in some places). Reference:
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_flatten

If we end up converting things to lodash, it becomes preferential to do what Foreman did w/ the lodash babel plugin. Went ahead and referenced that in [3776](https://github.com/ManageIQ/manageiq-ui-classic/pull/3776#issuecomment-402750123)
but happy to open a separate issue if we care at all about the performance of lodash.

Reply:
Love love the link, thanks for sharing!

Reply:
Ok mmm I would make one humble request, that we keep the naming consistent, renaming this to `flatten` would be nice 🤔

Reply:
sure, not a problem!